src/markup/codemirror/create.ts

```diff
@@ -152,7 +153,18 @@ export function createCodemirror(params: CreateCodemirrorParams) {
                 onScroll(event);
             },
             paste(event, editor) {
-                if (event.clipboardData && parseInsertedUrlAsImage) {
+                if (!event.clipboardData) return false;
+
+                const pastedText = event.clipboardData.getData('text/plain');
+
+                // Handle markdown-aware pasting
+                if (handleMarkdownPaste(pastedText, editor)) {
+                    event.preventDefault();
+                    return true;
+                }
```

Comment:
I'm guessing `handleMarkdownPaste()` should be after `parseInsertedUrlAsImage()`